components/Dialog/EditMetadata.vue

```diff
@@ -109,18 +135,32 @@ const isDate = (prop: any) => {
   return prop?.type == 'string' && prop?.format == 'date';
 };
 
+const isDateTime = (prop: any) => {
+  return prop?.type == 'string' && prop?.format == 'date-time';
+};
+
+const isSimpleString = (prop: any) => {
+  return prop?.type == 'string' && !prop?.format && !prop?.enum;
+};
+
 const isCheckbox = (prop: any) => {
   return prop?.type == 'boolean';
 };
 
 const updateMetadata = async () => {
+  const formattedMetadata: typeof metadata.value = {};
+  formattedMetadata.category = metadata.value.category;
```

Comment:
Here we assume that metadata fields can be or are `file`, `category`, `valid_from`.... but those field names are arbitrary

Actual types are defined in `collection.cmetadata.documents_metadata` - see `isDateTime`, `isSimpleString` etc methods